#!/bin/bash
#PBS -N SAMPLE_NAME       # Replace with your job name
#PBS -l nodes=1:ppn=1,walltime=20:00:00
#PBS -q bigmem
#PBS -S /bin/bash

# Change to the directory from which the job was submitted
cd $PBS_O_WORKDIR

# Run HipSTR: replace placeholders with your actual file paths
HipSTR \
    --bams    BAM_FILE.bam \      # Path to your sample BAM file
    --fasta   reference.fa \      # Path to your reference genome FASTA
    --regions regions.bed \       # Path to your STR regions BED file
    --str-vcf output.vcf.gz \     # Desired output VCF filename
    --viz-out output.viz.gz \     # Desired visualization output filename
    --min-reads 5 \               # Minimum supporting reads
    --def-stutter-mode            # Enable stutter mode

# Run dumpSTR: replace placeholders with your actual file paths and prefixes
dumpSTR \
    --vcf       INPUT.vcf.gz \        # Input VCF generated by HipSTR
    --out       OUTPUT_PREFIX \       # Output prefix for dumpSTR files
    --hipstr-min-call-Q 0.9 \         # Minimum call quality threshold
    --hipstr-max-call-flank-indel 0.15 \ # Maximum flank-indel ratio
    --hipstr-max-call-stutter 0.15    # Maximum stutter ratio

